modules/ROOT/pages/http/http-request-ref.adoc

```diff
@@ -36,6 +36,37 @@ By default, the HTTP connector limits the HTTP request header section size (requ
 
 Select Headers to add headers to the request. For example, add header names HeaderName1 and HeaderName2, add header values HeaderValue1 and HeaderValue2. You can use DataWeave expressions, for example `#[{'HeaderName1' : 'HeaderValue1', 'HeaderName2' : 'HeaderValue2'}]`.
 
+There are also use cases in which there are a set of default headers you always want to set. For example, you might want to always send authentication headers like `x-csrf-token: Fetch`.
+
+You can specify default headers at the config level so that you don't have to specify them on every single request. For example:
+
+[source,xml,linenums]
+----
+<http:request-config name="requestConfig">
+    <http:request-connection host="localhost" port="8081" streamResponse="true" responseBufferSize="1024"/>
+    <http:default-headers >
+        <http:default-header key="x-csrf-token" value="Fetch" />
+    </http:default-headers>
+</http:request-config>
+----
+
+With this configuration, those headers will be added to every outbound request, alongside any headers you specify in the actual request.
+
+The default headers also accept expressions, allowing you to use dynamic values. For example:
+
+[source,xml,linenums]
+----
+<http:request-config name="requestConfig">
+    <http:request-connection host="localhost" port="8081" streamResponse="true" responseBufferSize="1024"/>
+    <http:default-headers >
+        <http:default-header key="custom-role" value="#[vars.role]" />
+    </http:default-headers>
+</http:request-config>
+----
+
+This practice is helpful and convenient but has some important caveats to consider. Using expressions in a configuration element constitutes a dynamic configuration. The lifecycle of a dynamic configuration is explained here, but in a nutshell, all expressions in the config are evaluated each time the connector is used and for each set of distinct values a new configuration instance is created and initialized. 
+
+For the particular case of the HTTP connector, this means that it's only a good idea to use default headers with expressions when you are certain that the universe of all possible values that those expressions can take is small. If on the contrary, it is likely that every single evaluation is going to yield a different value, you'll end up creating several instances of the HTTP client. That will affect performance (due to the extra work that initializing a client takes), and it will be resource-consuming. For such cases, the recommendation is not to use default headers but to do it on the request operation itself.
```

Comment:
I'd recommend clarifying what "do it" means.

Reply:
done. Please check again